test/unittests/test_excel_to_json_resources.py

```diff
@@ -99,5 +101,36 @@ def test_excel2resources(self) -> None:
         self.assertListEqual(excel_first_class_cardinalities, json_first_class_cardinalities)
 
 
+    def test_validate_resources_with_schema(self) -> None:
+        # it is not possible to call the method to be tested directly. So let's make a reference to it, so that it can be found by the usage search
```

Comment:
If I right click on `def _validate_resources_with_schema(...)`, I can see where it is used all over the code. The idea is that if I am interested in where a method is used (or if I want to know if there is a unit test for a certain method), then I would like to find the unit test.
I must admit that my hack is a bit unorthodox. If you don't like it, I can remove it :-)

Reply:
I think the underlying issue is that you can't test the method directly. Why is that?

Reply:
Because at the beginning, we have an Excel file. Then it is processed by other methods (quite complicated processing), and the result is then sent to the Schema validation. This PR introduces a new feedback strategy of the schema validation: giving feedback that is relative to the initial Excel file, not the processed input that the schema validation receives. 

Now that I write this, I realize that this architecture is a bit suboptimal. I have one excuse, though: The only place where the schema validation is used, is here. So, the input will always originate from an Excel file. But I know that in the future, that might change.